src/Domain/BlockVector.hpp

```diff
@@ -0,0 +1,129 @@
+
+  // Member functions
+  using std::vector<T, Allocator>::vector;
+  using std::vector<T, Allocator>::operator=;
+  using std::vector<T, Allocator>::assign;
+  using std::vector<T, Allocator>::get_allocator;
+
+  // Element access
+  reference at(const BlockId& pos) noexcept {
+    return static_cast<std::vector<T, Allocator>&>(*this).at(pos.get_index());
+  }
+  const_reference at(const BlockId& pos) const noexcept {
+    return static_cast<const std::vector<T, Allocator>&>(*this).at(
+        pos.get_index());
+  }
+  reference operator[](const BlockId& pos) noexcept {
+    return static_cast<std::vector<T, Allocator>&>(*this)[pos.get_index()];
+  }
+  const_reference operator[](const BlockId& pos) const noexcept {
+    return static_cast<const std::vector<T, Allocator>&>(
+        *this)[pos.get_index()];
+  }
+  using std::vector<T, Allocator>::front;
+  using std::vector<T, Allocator>::back;
+  using std::vector<T, Allocator>::data;
+
+  // Iterators
+  using std::vector<T, Allocator>::begin;
+  using std::vector<T, Allocator>::cbegin;
+  using std::vector<T, Allocator>::end;
+  using std::vector<T, Allocator>::cend;
+
+  // Capacity
+  using std::vector<T, Allocator>::empty;
+  using std::vector<T, Allocator>::size;
+  using std::vector<T, Allocator>::max_size;
+  using std::vector<T, Allocator>::reserve;
+  using std::vector<T, Allocator>::capacity;
+  using std::vector<T, Allocator>::shrink_to_fit;
+
+  // Modifiers
+  using std::vector<T, Allocator>::clear;
+  using std::vector<T, Allocator>::insert;
+  using std::vector<T, Allocator>::emplace;
+  using std::vector<T, Allocator>::erase;
+  using std::vector<T, Allocator>::push_back;
+  using std::vector<T, Allocator>::emplace_back;
+  using std::vector<T, Allocator>::pop_back;
+  using std::vector<T, Allocator>::resize;
+  using std::vector<T, Allocator>::swap;
+
+  // Charm++ serialization
+  // NOLINTNEXTLINE(google-runtime-references)
+  void pup(PUP::er& p) noexcept {
+    p | static_cast<std::vector<T, Allocator>&>(*this);
+  }
+
+#define GENERATE_OP(OP)                                                       \
+  template <class S, class SAllocator = std::allocator<S>>                    \
+  friend bool operator OP(                                                    \
+      const domain::BlockVector<S, SAllocator>& lhs,                          \
+      const domain::BlockVector<S, SAllocator>& rhs) noexcept {               \
+    return static_cast<const std::vector<S, SAllocator>>(lhs) OP /* NOLINT */ \
+        static_cast<const std::vector<S, SAllocator>>(rhs);                   \
+  }
+  GENERATE_OP(==)
+  GENERATE_OP(!=)
+  GENERATE_OP(<)
+  GENERATE_OP(<=)
+  GENERATE_OP(>)
+  GENERATE_OP(>=)
+#undef GENERATE_OP
+
+  template <class S, class SAllocator = std::allocator<S>>
+  friend bool swap(const domain::BlockVector<S, SAllocator>& lhs,
+                   const domain::BlockVector<S, SAllocator>& rhs) noexcept {
```

Comment:
I don't see how this swap function could possibly work.  It takes its arguments by const reference.  Also it returns a bool which it obtains from the result of (std::?) swap, but that returns void.

Reply:
It was grabbing the swap inherited from `std::vector` on line 83. I've removed this implementation but left the test.